lgc/state/PipelineState.cpp

```diff
@@ -567,36 +576,43 @@ const ResourceNode *PipelineState::findPushConstantResourceNode() const {
 // @param binding : ID of descriptor binding
 std::pair<const ResourceNode *, const ResourceNode *>
 PipelineState::findResourceNode(ResourceNodeType nodeType, unsigned descSet, unsigned binding) const {
+  // Implements the <= order for the ResourceNodeType lattice.
+  // Returns true iff nodeType <= candidateType.
+  auto IsNodeTypeCompatible = [nodeType](ResourceNodeType candidateType) {
```

Comment:
camelBack isNodeTypeCompatible

Why do we want a lambda implicitly passing nodeType? Why not a simple function that passes both nodeType and candidateType?